boards/shared/Src/Io/Io_SharedSpi.c

```diff
@@ -41,92 +41,90 @@ void Io_SharedSpi_SetNssHigh(const struct SharedSpi *const spi_interface)
-        spi_interface->timeout_ms);
-    if (status != HAL_OK)
-    {
-        Io_SharedSpi_SetNssHigh(spi_interface);
-        return status;
-    }
 
-    status = HAL_SPI_Receive(
-        spi_interface->spi_handle, rx_buffer, rx_buffer_size,
-        spi_interface->timeout_ms);
-    if (status != HAL_OK)
+    if (HAL_SPI_Transmit(
+            spi_interface->spi_handle, tx_buffer, tx_buffer_size,
+            spi_interface->timeout_ms) == HAL_OK)
     {
-        Io_SharedSpi_SetNssHigh(spi_interface);
-        return (status);
+        status = HAL_SPI_Receive(
+                     spi_interface->spi_handle, rx_buffer, rx_buffer_size,
+                     spi_interface->timeout_ms) == HAL_OK;
     }
+
     Io_SharedSpi_SetNssHigh(spi_interface);
 
-    return HAL_OK;
+    return status;
 }
 
-HAL_StatusTypeDef Io_SharedSpi_Transmit(
+bool Io_SharedSpi_Transmit(
     const struct SharedSpi *const spi_interface,
     uint8_t *                     tx_buffer,
     uint16_t                      tx_buffer_size)
 {
     Io_SharedSpi_SetNssLow(spi_interface);
-    HAL_StatusTypeDef status = HAL_SPI_Transmit(
-        spi_interface->spi_handle, tx_buffer, tx_buffer_size,
-        spi_interface->timeout_ms);
+    const bool status =
+        HAL_SPI_Transmit(
+            spi_interface->spi_handle, tx_buffer, tx_buffer_size,
+            spi_interface->timeout_ms) == HAL_OK;
     Io_SharedSpi_SetNssHigh(spi_interface);
 
     return status;
 }
 
-HAL_StatusTypeDef Io_SharedSpi_Receive(
+bool Io_SharedSpi_Receive(
     const struct SharedSpi *const spi_interface,
     uint8_t *                     rx_buffer,
     uint16_t                      rx_buffer_size)
 {
     Io_SharedSpi_SetNssLow(spi_interface);
-    HAL_StatusTypeDef status = HAL_SPI_Receive(
-        spi_interface->spi_handle, rx_buffer, rx_buffer_size,
-        spi_interface->timeout_ms);
+    const bool status =
+        HAL_SPI_Receive(
+            spi_interface->spi_handle, rx_buffer, rx_buffer_size,
+            spi_interface->timeout_ms) == HAL_OK;
     Io_SharedSpi_SetNssHigh(spi_interface);
 
     return status;
 }
 
-HAL_StatusTypeDef Io_SharedSpi_MultipleTransmitWithoutNssToggle(
+bool Io_SharedSpi_MultipleTransmitWithoutNssToggle(
     const struct SharedSpi *const spi_interface,
     uint8_t *                     tx_buffer,
     uint16_t                      tx_buffer_size,
     size_t                        num_tx_data_copies)
 {
-    HAL_StatusTypeDef status;
+    bool status = true;
+
     for (size_t i = 0U; i < num_tx_data_copies; i++)
     {
-        status = HAL_SPI_Transmit(
-            spi_interface->spi_handle, tx_buffer, tx_buffer_size,
-            spi_interface->timeout_ms);
-        if (status != HAL_OK)
+        if (HAL_SPI_Transmit(
+                spi_interface->spi_handle, tx_buffer, tx_buffer_size,
+                spi_interface->timeout_ms) != HAL_OK)
         {
-            return status;
+            status = false;
+            break;
```

Comment:
There's no need for a status var, not like the status ever changes once it's set. Simpler to return false here and return true at the bottom of the method

Reply:
kinda wanted a single return statement here

Reply:
ehhhh im breaking anyways. sure i can just return false. will fix this

Reply:
i've seen bugs at work where people release a mutex after returning in a short function like this - big fan of one return per function to avoid stupid bugs like that